src/api/query.rs

```diff
@@ -90,13 +109,28 @@ pub fn block_hash_from_string(hash: &str) -> ApiResult<BlockHash> {
     })
 }
 
-pub fn account_address_from_identifier(id: &AccountIdentifier) -> ApiResult<AccountAddress> {
+pub enum Address {
```

Comment:
A comment explaining the semantics of the type, and why it exists, would be helpful..

Reply:
Done.